docs/contributing/release_process.rst

```diff
@@ -1,6 +1,15 @@
-Release Process
+Maintainer Guide
 ================
 
+Incoming Pull Request
+------------------------
+
+Any incoming Pull Request should be assigned to one or more maintainers for review. Upon approval, the PR should
+be **Squash and Merge**. If it's important to preserve a few commits during PR then **Rebase and merge** is acceptable.
+
```

Comment:
Maybe we should add a note about cleaning up the final squashed message if appropriate?